sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -656,6 +661,28 @@
       return toBuilder().setCommitOffsetsInFinalizeEnabled(true).build();
     }
 
+    /**
+     * Set additional parameters for KafkaUnboundedReader.offsetConsumer, added by BEAM-6285.
+     *
+     * <p>In KafkaIO.read(), there're two consumers running in the backend actually:<br>
+     * 1. KafkaUnboundedReader.consumer, which is the main consumer to read data from kafka
+     * topic(s);<br>
+     * 2. KafkaUnboundedReader.offsetConsumer, which is used to estimate backlog, by fetching the
+     * latest offset;<br>
+     *
+     * <p>By default, KafkaUnboundedReader.offsetConsumer shares the same parameters as
+     * KafkaUnboundedReader.consumer, with a generated ConsumerConfig.GROUP_ID_CONFIG(value is
+     * "%topicname_offset_consumer_%randonNumber_%groupId"). This may not work when the topic is
+     * secured in Kafka.
+     *
+     * <p>In this case with secured topics, you can add specific parameters for
+     * KafkaUnboundedReader.offsetConsumer, to overwrite parameters from
+     * KafkaUnboundedReader.consumer.
+     */
+    public Read<K, V> withOffsetConsumerConfig(Map<String, Object> offsetConsumerConfig) {
```

Comment:
Rename this to `withOffsetConsumerConfigOverrides()`, that way it is more explicit to the user that it first derives its config from the main consumer.

Reply:
+1